readthedocs/projects/tasks/builds.py

```diff
@@ -514,29 +515,13 @@ def on_failure(self, exc, task_id, args, kwargs, einfo):
         self.data.build['success'] = False
 
     def on_success(self, retval, task_id, args, kwargs):
-        html = self.data.outcomes['html']
-        search = self.data.outcomes['search']
-        localmedia = self.data.outcomes['localmedia']
-        pdf = self.data.outcomes['pdf']
-        epub = self.data.outcomes['epub']
-
-        time_before_store_build_artifacts = timezone.now()
-        # Store build artifacts to storage (local or cloud storage)
-        self.store_build_artifacts(
```

Comment:
we should probably do this change in another PR, doesn't look related to symlinks

Reply:
We need to move this call inside `execute` since we cannot raise exceptions from `on_success` and be able to manage them inside `on_failure` since it's not called when the code is already inside `on_success`.